pkg/controller/pipeline/pipeline_controller.go

```diff
@@ -322,6 +322,12 @@ func (r *ReconcileTektonPipeline) updateStatus(res *op.TektonPipeline, c op.Tekt
 	// to prevent us from using stale version of the object
 
 	tmp := res.DeepCopy()
+	for _, condition := range tmp.Status.Conditions {
+		if condition.Code == c.Code && condition.Version == c.Version {
+			return nil
+		}
+	}
+
```

Comment:
could you make this check `tmp.Status.Conditions[0].code == c.Code && ...`

otherwise, the loop will match the condition from the past, and skip a relevant update

Reply:
@nikhil-thomas  if we only check the [0], it will not work, since when reconcile is triggered, the new coming `condition` will always: 
`installing`
then
`installed`

Reply:
@vincent-pli oh, i see.. the status oscillates between `installing` and `install`. 
i think we can handle this better in the reconcile function it self.

could you move the check in this patch to the reconcile function (right before the `reconcileInstall` call).

ref: https://github.com/tektoncd/operator/blob/98cbaf7ed72855750c9d2b877f75515cc11d92f8/pkg/controller/pipeline/pipeline_controller.go#L193

Reply:
so `reconcileInstall` will not be called if the status is `installed` (the oscillating status updates won't be there)

Reply:
@nikhil-thomas Thanks for your reply.
I guess we should not block the `reconcileInstall`, besides the first installation, there is two kinds of events will trigger `reconcileInstall`.
- Any change on `TektonPipeline` and related `Deployment`, since we watch on them:
https://github.com/tektoncd/operator/blob/98cbaf7ed72855750c9d2b877f75515cc11d92f8/pkg/controller/pipeline/pipeline_controller.go#L112-L126
`Operator` try to handle the modification no matter it's on purpose or not.

- Regular event from apiserver (not sure, correct me if incorrect)

Reply:
i see, it makes sense.
:+1: